planning/behavior_path_planner/src/utils/goal_planner/default_fixed_goal_planner.cpp

```diff
@@ -38,12 +47,59 @@ BehaviorModuleOutput DefaultFixedGoalPlanner::plan(
   return output;
 }
 
+bool isInLanelets(const Pose & pose, const lanelet::ConstLanelets & lanes)
+{
+  for (const auto & lane : lanes) {
+    if (lanelet::utils::isInLanelet(pose, lane)) {
+      return true;
+    }
+  }
+  return false;
+}
+
+lanelet::ConstLanelets DefaultFixedGoalPlanner::extractLaneletsFromPath(
+  const PathWithLaneId & refined_path,
+  const std::shared_ptr<const PlannerData> & planner_data) const
+{
+  const auto & rh = planner_data->route_handler;
+  lanelet::ConstLanelets refined_path_lanelets;
+  for (size_t i = 0; i < refined_path.points.size(); ++i) {
+    const PathPointWithLaneId & path_point = refined_path.points[i];
+    int64_t lane_id = path_point.lane_ids[0];
+    lanelet::ConstLanelet lanelet = rh->getLaneletsFromId(lane_id);
+    bool is_unique = true;
+    for (const lanelet::ConstLanelet & existing_lanelet : refined_path_lanelets) {
+      if (lanelet == existing_lanelet) {
+        is_unique = false;
+        break;
+      }
+    }
+    if (is_unique) {
+      refined_path_lanelets.push_back(lanelet);
+    }
+  }
+  return refined_path_lanelets;
+}
+
+bool DefaultFixedGoalPlanner::isPathValid(
+  const PathWithLaneId & refined_path,
+  const std::shared_ptr<const PlannerData> & planner_data) const
+{
+  const lanelet::ConstLanelets lanelets = extractLaneletsFromPath(refined_path, planner_data);
+  for (size_t i = 0; i < refined_path.points.size(); ++i) {
+    const PathPointWithLaneId & path_point = refined_path.points[i];
```

Comment:
```suggestion
    const PathPointWithLaneId & path_point = refined_path.points[i];
```
```suggestion
    const PathPointWithLaneId & path_point = refined_path.points.at(i);
```

Reply:
changed to auto [https://github.com/autowarefoundation/autoware.universe/commit/9a198afc55c786c885f9d03f6b24349a4cee3186](https://github.com/autowarefoundation/autoware.universe/commit/9a198afc55c786c885f9d03f6b24349a4cee3186)